security/src/main/java/com/fincity/security/dao/UserDAO.java

```diff
@@ -409,4 +414,114 @@ public Mono<Set<ULong>> getUserListFromClientIds(Set<ULong> clientList) {
 
 		return Mono.just(users);
 	}
+
+	public Mono<Boolean> checkPasswordEqual(ULong userId, String newPassword) {
+
+		return Mono.from(
+
+		        this.dslContext.select(SECURITY_USER.PASSWORD)
+		                .from(SECURITY_USER)
+		                .where(SECURITY_USER.ID.eq(userId))
+		                .limit(1))
+		        .map(Record1::value1)
+		        .map(pass -> pass.equals(newPassword));
+
+	}
+
+	public Mono<Set<String>> getPastPasswords(ULong userId) {
+
+		return Flux.from(
```

Comment:
This can't be a set of strings. You need a list with the password types, too. If it is hashed, then you need to check differently. Please refer to Authentication Service.

Reply:
updated